tests/pass/move-data-across-await-point.rs

```diff
@@ -0,0 +1,84 @@
+use std::future::Future;
+
+// This test:
+// - Compares addresses of non-Copy data before and after moving it
+// - Writes to the pointer after it has moved across the await point
+//
+// This is only meant to assert current behavior, not guarantee that this is
+// how it should work in the future. In fact, upcoming changes to rustc
+// *should* break these tests.
+// See: https://github.com/rust-lang/rust/issues/62958
+async fn data_moved_async() {
+    async fn helper(mut data: Vec<u8>, raw_pointer: *mut Vec<u8>) {
+        let raw_pointer2 = &mut data as *mut Vec<u8>;
+        // When the Vec is moved into the generator upvar, that copies the Vec's
+        // ptr+capacity+len that is on the stack. Because of that, the value of
+        // `raw_pointer2` is different than that of `raw_pointer`.
```

Comment:
```suggestion
        // `raw_pointer` points to the original location where the Vec was stored in the caller.
        // `data` is where that Vec (to be precise, its ptr+capacity+len on-stack data)
        // got moved to. Those will usually not be the same since the Vec got moved twice
        // (into the function call, and then into the generator upvar).
```